shell/platform/android/io/flutter/view/AccessibilityViewEmbedder.java

```diff
@@ -470,32 +496,102 @@ private Long getSourceNodeId(@NonNull AccessibilityNodeInfo node) {
+                // catch statements, or catch Exception. [NewApi]
+                } catch (IllegalAccessException e) {
+                    Log.w(TAG, e);
+                } catch (InvocationTargetException | ArrayIndexOutOfBoundsException e) {
+                    Log.w(TAG, e);
+                }
             }
             return null;
         }
 
         @Nullable
         private Long getParentNodeId(@NonNull AccessibilityNodeInfo node) {
-            if (getParentNodeId == null) {
+            if (getParentNodeId != null) {
+                try {
+                    return (long) getParentNodeId.invoke(node);
+                // Using identical separate catch blocks to comply with the following lint:
+                // Error: Multi-catch with these reflection exceptions requires API level 19
+                // (current min is 16) because they get compiled to the common but new super
+                // type ReflectiveOperationException. As a workaround either create individual
+                // catch statements, or catch Exception. [NewApi]
+                } catch (IllegalAccessException e) {
+                    Log.w(TAG, e);
+                } catch (InvocationTargetException e) {
+                    Log.w(TAG, e);
+                }
+            }
+
+            // Fall back on reading the ID from a serialized data if we absolutely have to.
+            return yoinkParentIdFromParcel(node);
+        }
+
+        @Nullable
+        private static Long yoinkParentIdFromParcel(AccessibilityNodeInfo node) {
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+                Log.w(TAG, "Unexpected Android version. Unable to find the parent ID.");
                 return null;
             }
-            try {
-                return (long) getParentNodeId.invoke(node);
-            } catch (IllegalAccessException e) {
-                Log.w(TAG, e);
-            } catch (InvocationTargetException e) {
-                Log.w(TAG, e);
+
+            // If this looks like it's failing, that's because it probably is. This method is
+            // relying on the implementation details of `AccessibilityNodeInfo#writeToParcel` in
+            // order to find the particular bit in the opaque parcel that represents mParentNodeId.
+            // If these implementation details change from our assumptions in this method, this will
+            // silently break.
+            AccessibilityNodeInfo copy = AccessibilityNodeInfo.obtain(node);
+            final Parcel parcel = Parcel.obtain();
+            parcel.setDataPosition(0);
+            copy.writeToParcel(parcel, /*flags=*/ 0);
+            Long parentNodeId = null;
+            // Match the internal logic that sets where the data actually ends up finally living.
+            // Move through the parcel if and when there's data set to read so that we can get to
+            // where the parentNodeId has been stored.
+            parcel.setDataPosition(0);
```

Comment:
nit: let's point in a comment to the logic this is matching (https://android.googlesource.com/platform/frameworks/base/+/0b5ca24a41ec6fe7fe34583cbb7871886f6bf25e/core/java/android/view/accessibility/AccessibilityNodeInfo.java#3315 )